### PR TITLE
Add learning rate schedule example

### LINKED ISSUE #246: Can you provide an example how to use lr_schedule  in stable baselines 3?
I am migrating from stable baselines to SB3 and I understand that LinearSchedule is no longer supported?
It is not in the docs.

### PR DESCRIPTION
**Pull request by araffin**
## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

closes #246 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [x] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [x] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)
- [x] I have checked that the documentation builds using `make doc` (**required**)

Note: You can run most of the checks using `make commit-checks`.

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

File: docs/guide/examples.rst [hunk 1]
```diff
@@ -422,6 +422,53 @@ The parking env is a goal-conditioned continuous control task, in which the vehi
           obs = env.reset()
 
 
+Learning Rate Schedule
+----------------------
+
+All algorithms allow to pass a learning rate schedule that takes as input the current progress remaining (from 1 to 0).
```
Comment (AdamGleave) [inline docs/guide/examples.rst:428]:
```suggestion
All algorithms allow you to pass a learning rate schedule that takes as input the current progress remaining (from 1 to 0).
```
Or "allow one".

File: docs/guide/examples.rst [hunk 2]
```diff
@@ -422,6 +422,53 @@ The parking env is a goal-conditioned continuous control task, in which the vehi
...
+          :return: current learning rate
+          """
+          return progress_remaining * initial_value
+
+      return func
+
+  # Initial learning rate of 0.001
+  model = PPO("MlpPolicy", "CartPole-v1", learning_rate=linear_schedule(0.001), verbose=1)
+  model.learn(total_timesteps=20000)
+  # On the second call, by default, the learning rate schedule resets
```
Comment (AdamGleave) [inline docs/guide/examples.rst:467]: I find it a bit confusing that we have "by default" here and then manually specify `reset_num_timesteps` to True. I'd suggest either dropping the keyword argument, or clarifying the comment: "By default, `reset_num_timesteps` is True, in which case the learning rate schedule resets."

**Review by AdamGleave — CHANGES_REQUESTED** [review]
Thanks for adding the example -- couple of changes suggested to improve clarity, but mostly LGTM

**Review by AdamGleave — APPROVED** [review]
LGTM, thanks for the changes

**Comment by araffin** [comment]
>LGTM, thanks for the changes

Thanks for the review =) (next time you can merge it once you agree with the changes ;) )
